apps/aecore/src/aec_chain_state.erl

```diff
@@ -106,8 +109,19 @@ top_block(?match_state(top_block_hash := X) = State) ->
     export_block(blocks_db_get(X, State), State).
 
 -spec insert_block(#block{}, state()) -> {'ok', state()} | {'error', any()}.
-insert_block(Block, ?assert_state() = State) ->
-    try {ok, internal_insert(wrap_block(Block), State)}
+insert_block(Block, ?assert_state() = State0) ->
+    try 
+        State1 = internal_insert(wrap_block(Block), State0),
+        State =
+            case Block#block.height of
+                0 ->
+                    {ok, GenesisHash} =
+                        aec_blocks:hash_internal_representation(Block),
+                    State1#{genesis_block_hash => GenesisHash};
```

Comment:
Is this line changing the genesis???

Reply:
I see handle_call(write_block) then ends here.

Reply:
if a new genesis block is added - it gets rejected in the checks in aec_chain_state:internal_insert/2 the check itself is done [here](https://github.com/velzevur/epoch/blob/7cf6f7345cdd0fe95269a68407f79fa1af986a73/apps/aecore/src/aec_chain_state.erl#L423)
anyway - I've added an assertion